rollup/fees/rollup_fee.go

```diff
@@ -150,30 +169,8 @@ func mulAndScale(x *big.Int, y *big.Int, precision *big.Int) *big.Int {
 	return new(big.Int).Quo(z, precision)
 }
 
-// copyTransaction copies the transaction, removing the signature
-func copyTransaction(tx *types.Transaction) *types.Transaction {
-	if tx.To() == nil {
-		return types.NewContractCreation(
-			tx.Nonce(),
-			tx.Value(),
-			tx.Gas(),
-			tx.GasPrice(),
-			tx.Data(),
-		)
-	}
-	return types.NewTransaction(
-		tx.Nonce(),
-		*tx.To(),
-		tx.Value(),
-		tx.Gas(),
-		tx.GasPrice(),
-		tx.Data(),
-	)
-}
-
 func CalculateFees(tx *types.Transaction, state StateDB) (*big.Int, *big.Int, *big.Int, error) {
```

Comment:
Add `l2Fee` and `fee` used anywhere?

Reply:
you mean rename the func to `CalculateL1DataFees`?

Reply:
For `L1MessageTx` tx, we should directly return 0 in the case of tracing such tx.

Reply:
I think peter means "Add `l2Fee` and `fee` used anywhere?"

`l2Fee` is used in `fees.VerifyFee()`

So, should I base on `relay-l1-messages` branch? Currently `L1MessageTx` is not there on `develop` yet.

Or I can add this logic shortly, after https://github.com/scroll-tech/go-ethereum/pull/350 is merged (which seems soon).

Reply:
Maybe rename this function to `CalculateTxL1DataFee` to be more precise? alternatively, we could make the current `CalculateL1DataFee` to an internal function eg. `calculateFee`, and make this function `CalculateL1DataFee`

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/pull/351/commits/aac08fbfd05fa8a94b69a90fa8b9d83d165d1b3d